File: backend/logsheet/service/MenuItemService.java
package com.sunbeam.logsheet.service;

import java.util.List;

import com.sunbeam.logsheet.DTO.ApiResponse;
import com.sunbeam.logsheet.DTO.MenuItemRequestDTO;
import com.sunbeam.logsheet.DTO.MenuItemResponseDTO;

public interface MenuItemService {

    ApiResponse<?> createMenuItem(MenuItemRequestDTO dto);

    ApiResponse<?> updateMenuItem(Long id, MenuItemRequestDTO dto);

    ApiResponse<?> deleteMenuItem(Long id);

    List<MenuItemResponseDTO> getAllMenuItems();

    MenuItemResponseDTO getMenuItemById(Long id);
}

File: frontend/logsheet_fe/src/services/MenuItemService.js


import api from "../utils/api";


export const getAllMenuItems = () => api.get("/menu-items");
export const getMenuItemById = (id) => api.get(`/menu-items/${id}`);
export const createMenuItem = (data) => api.post("/menu-items", data);
export const updateMenuItem = (id, data) => api.put(`/menu-items/${id}`, data);
export const deleteMenuItem = (id) => api.delete(`/menu-items/${id}`);
